frontend/online-shop/src/components/product-details/products-details.component.tsx

```diff
@@ -0,0 +1,56 @@
+import { Button, Typography } from "@mui/material";
+
+import { FC, useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { fetchProductById } from "../../services/products.service";
+
+const ProductDetail: FC = () => {
+  let params = useParams();
+
+  const [product, setProduct] = useState({
```

Comment:
I would suggest to leave it as `undefined` as the initial value and within the application show a loading spinner while it's undefined